UI/script.js

```diff
@@ -0,0 +1,354 @@
+                        }
+                    }
+                    break;
+                case 'discount':
+                    if (typeof ad.discount !== 'string' || ad.discount.length === 0) {
+                        return false;
+                    }
+                    break;
+                case 'validUntil':
+                    if (!(ad.validUntil instanceof Date)) {
+                        return false;
+                    }
+                    break;
+                case 'rating':
+                    if (!(typeof ad.rating === 'undefined' || typeof ad.rating === 'number')){
+                        return false;
+                    }
+                    break;
+                case 'reviews':
+                    if (!Array.isArray(ad.reviews)) {
+                        return false;
+                    }
+                    break;
+                default:
+                    return false;
+            }
+        }
+        return true;
+    }
+
+    add(ad){
+        if (AdsList.validate(ad)) {
+            this._adList.push(ad);
+            return true;
+        }
+        return false;
+    }
+
+    remove(id){
+        if (typeof id === 'string'){
+            let index = this._adList.findIndex(ad => ad.id === id);
+            if (index !== -1){
+                this._adList.splice(index, 1);
+    
+                return true;
+            }
+        }
+        return false;
+    }
+
+    edit(id, ad){
+        for (let param in ad){
+            if (param === 'id' || param === 'vendor' || param === 'createdAt'){
+                console.log("You can't change id, vendor, createdAt");
+                return false;
+            }
+        }
+        
+        var tmpAd = {};
+        Object.assign(tmpAd, this.get(id));
+        
+        for (let param in ad){
+            tmpAd[param] = ad[param];
+        }
+    
+        if (!AdsList.validate(tmpAd)){
+            return false;
+        }
+        
+        this.remove(id);
+        this.add(tmpAd);
+
+        return true;
+    }
+
+    addAll(adList){
+        return adList.filter(ad => !this.add(ad));
+    }
+    
+    clear(){
+        this._adList.splice(0, this._adList.length);
+    }
+}
+
+class View{
+    _ads;
+    _tmpAd;
+    _username;
+    _isVendor;
+    constructor(adList, username, isVendor){
+        this._ads = new AdsList(adList);
+        this._tmpAd = document.querySelector(".offer-tmp");
+        this._username = username;
+        this._isVendor = isVendor;
+    }
+    showUser(){
+        if(this._username == null || this._username.length == 0){
+            document.querySelector(".profile").textContent = "Log in";
+            document.querySelector(".log-out").style.visibility = "hidden";
+        }
+        else {
+            document.querySelector(".profile").textContent = this._username;
+            document.querySelector(".log-out").style.visibility = "visible";
+        }
+    }
+    _getOffer(ad){
```

Comment:
I think it shouldn't be called `getOffer`, it should be something like `buildOffer`